### PR TITLE
[MAINTENANCE] A small refactor of ParamerBuilder management used in DataAssistant classes

### PR DESCRIPTION
**Pull request by alexsherstinsky**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-761/GREAT-866
- JIRA: GREAT-761/GREAT-869
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/rule_based_profiler/parameter_builder/parameter_builder.py [hunk 1]
```diff
@@ -267,6 +267,10 @@ def evaluation_parameter_builder_configs(
     def json_serialize(self) -> Union[str, bool]:
         return self._json_serialize
 
+    @json_serialize.setter
+    def json_serialize(self, value: Union[str, bool]):
+        self._json_serialize = value
```
Comment (cdkini) [inline great_expectations/rule_based_profiler/parameter_builder/parameter_builder.py]: Please annotates as `-> None` 🙇🏽

File: great_expectations/rule_based_profiler/data_assistant/data_assistant.py [hunk 2]
```diff
@@ -85,6 +89,57 @@ class DataAssistant(metaclass=MetaDataAssistant):
         ...
     """
 
+    COMMONLY_USED_PARAMETER_BUILDERS: Dict[str, ParameterBuilder] = {
+        "table_row_count_metric_multi_batch_parameter_builder": MetricMultiBatchParameterBuilder(
```
Comment (cdkini) [inline great_expectations/rule_based_profiler/data_assistant/data_assistant.py]: Same note as a prior PR about magic strings - could we avoid hardcoding this? Or perhaps uses constants or an enum? Between this dict and the self-initializing expectation configs, this becomes hard to maintain.

I think we could encapsulate these in a class and access through dot notation?

**Review by cdkini — CHANGES_REQUESTED** [review]

File: great_expectations/rule_based_profiler/data_assistant/data_assistant.py (hunk 2 above)
Comment (cdkini) [reply great_expectations/rule_based_profiler/data_assistant/data_assistant.py]: On second thought, this is just internally used right? I feel less iffy about it then.

Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/data_assistant/data_assistant.py]: Yes, only internally.

**Review by cdkini — APPROVED** [review]

File: great_expectations/rule_based_profiler/data_assistant/data_assistant.py (hunk 2 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/data_assistant/data_assistant.py]: @cdkini This has been addressed -- please take a look.  Thanks!

File: great_expectations/rule_based_profiler/parameter_builder/parameter_builder.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/parameter_builder/parameter_builder.py]: @cdkini fixed -- thanks!

**Pull request merged**